forge-game/src/main/java/forge/game/ability/effects/DigUntilEffect.java

```diff
@@ -135,6 +135,10 @@ public void resolve(SpellAbility sa) {
         boolean sequential = digSite == ZoneType.Library && revealedDest != null && revealedDest.equals(foundDest);
 
         CardZoneTable table = new CardZoneTable(game.copyLastStateBattlefield(), game.copyLastStateGraveyard());
+        CardZoneTable tableSeq = null;
+        if (!sequential) {
+            tableSeq = new CardZoneTable(table.getLastStateBattlefield(), table.getLastStateGraveyard());
+        }
```

Comment:
reuse table if not sequential?

`CardZoneTable tableSeq = sequential ? null : table;`

Reply:
I can't do that currently

because even when revealing first the found ones get moved together before the other ones
e. g. on _Collision of Realms_:
> All creatures are shuffled into their owners' libraries at the same time. Then all players who shuffled nontoken creatures into their library this way reveal cards. **Then all creature cards revealed this way enter the battlefield at the same time. Then all other cards are put on the bottom of their owners' libraries.**

It would need more refactoring to split the Players loop into two first but that might get a bit messy with some of the checks on those cards

and I'm not sure a corner case for it exists anyway... 🤔

Reply:
if not sequential both will be used
`table` will contain the revealed cards that should be moved after doing the found ones